src/System.Linq/tests/SingleTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Linq.Tests.Helpers;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class SingleTests
+    {
+        [Fact]
+        public void FailOnEmpty()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Single());
+        }
+
+        [Fact]
+        public void DefaultOnEmpty()
+        {
+            Assert.Equal(0, Enumerable.Empty<int>().SingleOrDefault());
+        }
+
+        [Fact(Skip="Runs for thousands of years. Then throws wrong exception")]
+        public void FailOnManyMatches()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Range(0, int.MaxValue).Single(i => i > 0));
```

Comment:
Got you. I was more thinking that adding them now is premature because I'm more used to workflows where one doesn't test for behaviour one hasn't decided is or isn't desirable. I'll add them to this PR later (one a phone right now).